subprojects/packagefiles/cmocka/meson.build

```diff
@@ -53,46 +53,7 @@ config = configuration_data()
 
 if compiler.get_argument_syntax() == 'gcc'
   add_project_arguments(
-# I've explicitly skipped the duplicated -W versions when they also test
-# for the -Werror version
-    compiler.get_supported_arguments(
-      '-Wshadow',
-      '-Wmissing-prototypes',
-      '-Wcast-align',
-      '-Werror=address',
-      '-Werror=strict-prototypes',
-      '-Werror=write-strings',
-      '-Werror=implicit-function-declaration',
-      '-Werror=pointer-arith',
-      '-Werror=declaration-after-statement',
-      '-Werror=return-type',
-      '-Werror=uninitialized',
-      '-Wimplicit-fallthrough',
-      '-Werror=strict-overflow',
-      '-Wstrict-overflow=2',
-      '-Wno-format-zero-length',
-      '-Wformat',
-      '-Werror=format-security',
-      '-Wno-gnu-zero-variadic-macro-arguments',
-      '-D_GNU_SOURCE',
-    ),
-    language: 'c',
-    native: is_native,
-  )
-  # We can't test the build type, so we can' add -D_FORTIFY_SOURCE=2 here
-  if machine.system() == 'darwin'
-    add_project_arguments(
-      compiler.get_supported_arguments('-Wno-deprecated-declarations'),
-      language: 'c',
-      native: is_native,
-    )
-  endif
-else
-  add_project_arguments(
-    '/D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1',
-    '/D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES_COUNT=1',
-    '/D_CRT_NONSTDC_NO_WARNINGS=1',
-    '/D_CRT_SECURE_NO_WARNINGS=1',
```

Comment:
This one is still useful on clang-cl.

Reply:
That's -Wdeprecated-declarations, no?

Reply:
Sure, it's a bigger hammer but that'd work too.